### PR TITLE
Integrate SQ in Bitbucket pipeline

### PR DESCRIPTION
**Pull request by amelie-renard-sonarsource**

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -1,3 +1,21 @@
+= C++ example project scanned on SonarQube using Bitbucket Cloud pipelines
+
+To analyze a C/C++/Objective-C project on Bitbucket pipelines with SonarQube, you need to:
+
+. Enable pipelines in your Bitbucket Cloud repository
+. Add a repository variable `SONAR_TOKEN` with a token generated in SonarQube (see https://docs.sonarqube.org/latest/user-guide/user-token/[Generating and Using Tokens])
+. Add a repository variable `SONAR_HOST_URL` that points to your SonarQube server
+. In your `bitbucket-pipelines.yml` file:
+.. Download the https://sonarcloud.io/documentation/analysis/languages/cfamily/#analysis-steps-using-build-wrapper[`build wrapper`] and wrap your compilation with it
```
Comment (arseniy-sonar) [inline README.adoc]: Do you really want a SonarCloud link in a SonarQube-oriented repository?

File: bitbucket-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+pipelines:
+  branches:
+    master:
+      - step:
+          name: Build and analyze on SonarQube
+          script:
+            - curl --create-dirs -sSLo $HOME/.sonar/build-wrapper-linux-x86.zip ${SONAR_HOST_URL}/static/cpp/build-wrapper-linux-x86.zip
+            - unzip -o $HOME/.sonar/build-wrapper-linux-x86.zip -d $HOME/.sonar/
+            - export PATH=$HOME/.sonar/build-wrapper-linux-x86:$PATH
+            - export BW_OUTPUT=bw-output
```
Comment (arseniy-sonar) [inline bitbucket-pipelines.yml]: Is there a way to move this out of the script into a yml field? Maybe `env:` or `variables:`?

File: README.adoc [hunk 3]
```diff
@@ -1,3 +1,21 @@
+= C++ example project scanned on SonarQube using Bitbucket Cloud pipelines
+
+To analyze a C/C++/Objective-C project on Bitbucket pipelines with SonarQube, you need to:
+
+. Enable pipelines in your Bitbucket Cloud repository
+. Add a repository variable `SONAR_TOKEN` with a token generated in SonarQube (see https://docs.sonarqube.org/latest/user-guide/user-token/[Generating and Using Tokens])
+. Add a repository variable `SONAR_HOST_URL` that points to your SonarQube server
```
Comment (arseniy-sonar) [inline README.adoc]: add an example (e.g., https://example.com:9000) ?

File: bitbucket-pipelines.yml (hunk 2 above)
Comment (arseniy-sonar) [reply bitbucket-pipelines.yml]: Also, to keep consistent with other examples and be explicit, I would spell `build_wrapper_output_directory` in full

**Review by arseniy-sonar — COMMENTED** [review]
Looks good. Just a few comments

File: README.adoc (hunk 1 above)
Comment (amelie-renard-sonarsource) [reply README.adoc]: No, good catch

File: bitbucket-pipelines.yml (hunk 2 above)
Comment (amelie-renard-sonarsource) [reply bitbucket-pipelines.yml]: Unfortunately no, I did not find a way to move the variables in a different section

**Pull request merged**